Update task-6 to use sock, instead of conn

Task-06 still uses the - now deprecated - conn API, which should be updated to sock. Further, the README show the shell command `txtsnd` which is not present anymore, see #26.

btw. `conn` is actually removed already from RIOT
```

/Volumes/workspace/github/RIOT-OS/Tutorials/task-06/udp.c:6:10: fatal error: 'net/conn/udp.h' file not found
#include "net/conn/udp.h"
         ^

```

--- a/task-06/Makefile
+++ b/task-06/Makefile
@@ -26,6 +26,6 @@ USEMODULE += gnrc_netdev_default
 USEMODULE += auto_init_gnrc_netif
 USEMODULE += gnrc_ipv6_default
 USEMODULE += gnrc_icmpv6_echo
-USEMODULE += gnrc_conn_udp
+USEMODULE += gnrc_sock_udp
 
 include $(RIOTBASE)/Makefile.include

--- a/task-06/README.md
+++ b/task-06/README.md
@@ -2,16 +2,16 @@
 
 # 6. UDP Client / Server
 
-The transport layer (UDP, TCP, etc) is accessed through [`conn`](https://doc.riot-os.org/group__net__conn.html) driver API
+The transport layer (UDP, TCP, etc) is accessed through [`sock`](https://doc.riot-os.org/group__net__sock.html) driver API
 
 ![Networking overview](../overview-net.png)
 
 Look at the new modules in the [`Makefile`](Makefile)
 ```
-USEMODULE += gnrc_conn_udp
+USEMODULE += gnrc_sock_udp
 ```
 
-[`udp.c`](udp.c) utilizes `conn_udp_sendto()` and `conn_udp_recvfrom()` to exchange UDP packets
+[`udp.c`](udp.c) utilizes `sock_udp_send()` and `sock_udp_recv()` to exchange UDP packets
 
 ## Task 6.1: Use UDP for messaging
 1.  Compile and run on two `native` instances
@@ -90,7 +90,7 @@ USEMODULE += gnrc_conn_udp
     ```
     udp <tap0-IPv6-addr> 8888 hello
     ```
-[Read the Doc](https://doc.riot-os.org/group__net__conn.html)
+[Read the Doc](https://doc.riot-os.org/group__net__sock.html)
 
 ## Task 6.3 -- Exchange UDP packets with your neighbors
 * Compile, flash and run on the board `BOARD=samr21-xpro make all flash term`

--- a/task-06/udp.c
+++ b/task-06/udp.c
@@ -2,8 +2,7 @@
 #include <stdint.h>
 #include <stdio.h>
 
-#include "net/af.h"
-#include "net/conn/udp.h"
+#include "net/sock/udp.h"
 #include "net/ipv6/addr.h"
 #include "thread.h"
 
@@ -11,30 +10,29 @@
 #define SERVER_BUFFER_SIZE      (64)
 
 static bool server_running = false;
-static conn_udp_t conn;
+static sock_udp_t sock;
 static char server_buffer[SERVER_BUFFER_SIZE];
 static char server_stack[THREAD_STACKSIZE_DEFAULT];
 static msg_t server_msg_queue[SERVER_MSG_QUEUE_SIZE];
 
 void *_udp_server(void *args)
 {
-    uint16_t port = (uint16_t) atoi(args);
-    ipv6_addr_t server_addr = IPV6_ADDR_UNSPECIFIED;
+    sock_udp_ep_t server = { .port = atoi(args), .family = AF_INET6 };
     msg_init_queue(server_msg_queue, SERVER_MSG_QUEUE_SIZE);
 
-    if(conn_udp_create(&conn, &server_addr, sizeof(server_addr), AF_INET6, port) < 0) {
+    if(sock_udp_create(&sock, &server, NULL, 0) < 0) {
         return NULL;
     }
 
     server_running = true;
-    printf("Success: started UDP server on port %" PRIu8 "\n", port);
+    printf("Success: started UDP server on port %u\n", server.port);
 
     while (1) {
         int res;
-        ipv6_addr_t src;
-        size_t src_len = sizeof(ipv6_addr_t);
-        if ((res = conn_udp_recvfrom(&conn, server_buffer, sizeof(server_buffer) - 1,
-                                     &src, &src_len, &port)) < 0) {
+
+        if ((res = sock_udp_recv(&sock, server_buffer,
+                                 sizeof(server_buffer) - 1, SOCK_NO_TIMEOUT,
+                                 NULL)) < 0) {
             puts("Error while receiving");
         }
         else if (res == 0) {
@@ -51,20 +49,20 @@ void *_udp_server(void *args)
 
 int udp_send(int argc, char **argv)
 {
+    int res;
+    sock_udp_ep_t remote = { .family = AF_INET6 };
+
     if (argc != 4) {
         puts("Usage: udp <ipv6-addr> <port> <payload>");
         return -1;
     }
 
-    int res;
-    ipv6_addr_t src = IPV6_ADDR_UNSPECIFIED, dst;
-    if (ipv6_addr_from_str(&dst, argv[1]) == NULL) {
+    if (ipv6_addr_from_str((ipv6_addr_t *)&remote.addr, argv[1]) == NULL) {
         puts("Error: unable to parse destination address");
         return 1;
     }
-
-    if((res = conn_udp_sendto(argv[3], strlen(argv[3]), &src, sizeof(src), &dst, sizeof(dst),
-                              AF_INET6, 1234, (uint16_t) (atoi(argv[2])))) < 0) {
+    remote.port = atoi(argv[2]);
+    if((res = sock_udp_send(NULL, argv[3], strlen(argv[3]), &remote)) < 0) {
         puts("could not send");
     }
     else {

